packages/engine.test/src/testProducer.ts

```diff
@@ -0,0 +1,104 @@
+expect.extend({
+    toMatchExpectations(mockCalls, verbExpectations) {
+        // const remainingExpectations = verbExpectations.slice(0)
+        // const remainingMockCalls = mockCalls.slice(0)
+    
+        const isLonger = mockCalls.length !== verbExpectations.length
+
+        if(isLonger) return {
+            message: () => (`expected calls ${this.utils.printReceived(mockCalls)} to be same length as expectation ${this.utils.printExpected(verbExpectations)}`),
+            pass: false
+        }
+
+        let isSame: boolean = true
+        for(let i = 0; i < mockCalls.length; i++) {
+            expect(mockCalls[i][0]).toEqual(verbExpectations[i])
+
+            // if(verbExpectations[i].match) {
+            //     expect(mockCalls[i][0]).toMatchObject(verbExpectations[i].match)
+            // }
+            // if(verbExpectations[i].equal) {
+            //     expect(mockCalls[i][0]).toEqual(verbExpectations[i].equal)
+            // }
+        }
+
+        // const pass = true
+      
+  
+        // if (pass) {
+        //     return {
+        //         message: () => (`expected ${this.utils.printReceived(received)} not to contain object ${this.utils.printExpected(argument)}`),
+        //         pass: true
+        //     }
+        // } else {
+        //     return {
+        //         message: () => (`expected ${this.utils.printReceived(received)} to contain object ${this.utils.printExpected(argument)}`),
+        //         pass: false
+        //     }
+        // }
+    }
+  })
+
+jest.useFakeTimers();
+
+export function testProducer(name: string, {
```

Comment:
Make the function polymorphic `testProducer(name, config || fn)`.

The fn should return a config object.

This way developers can compose config inside their tests.